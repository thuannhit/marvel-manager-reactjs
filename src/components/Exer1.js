import React, {Component} from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
class Ex1 extends Component{
    RegExp=""
    constructor(props){
        super(props);
        this.state={
            result:'',
            inputString:''
        }
    }
    onClickRemoveVowels =(oEvent)=>{
        var sInputString=this.state.inputString;
        this.setState({
            result: (sInputString) ? sInputString.replace(/[aeiou]/gi, ''): ''
        })
        console.log("I am removing vowels");
    }
    handleChange=(oEvent)=>{
        this.setState({inputString:oEvent.target.value})
        console.log("It is changing somethings");
    }
    render(){
        let sResult=this.state.result;
        return (
            <div className="d-flex flex-column">
                <br/>
                <TextField id="standard-full-width" label="Input string" style={{ margin: 8 }}
                    placeholder="Your string"
                    helperText="After typing your string, please press Remove vowels button"
                    fullWidth
                    margin="normal"
                    value={this.state.inputString}
                    onChange={this.handleChange}/>
                <Button className="Button" width="50px" onClick={this.onClickRemoveVowels.bind(this)}>Remove vowels</Button>
                <p>Result: </p><p style={{fontWeight: "bold"}}>{sResult}</p>
            </div>
        )
    }
}

export default Ex1;