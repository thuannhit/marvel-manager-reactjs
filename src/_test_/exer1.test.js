import React from 'react';
import Exer1 from '../components/Exer1';
import renderer from 'react-test-renderer';

it('Exer1 renders correctly', () => {
  const tree = renderer
    .create(<Exer1/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});