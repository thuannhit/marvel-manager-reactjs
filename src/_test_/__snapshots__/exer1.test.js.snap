// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exer1 renders correctly 1`] = `
<div
  className="d-flex flex-column"
>
  <br />
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <label
      className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="standard-full-width"
    >
      Input string
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-describedby="standard-full-width-helper-text"
        aria-invalid={false}
        className="MuiInputBase-input-46 MuiInput-input-31"
        disabled={false}
        id="standard-full-width"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Your string"
        required={false}
        type="text"
        value=""
      />
    </div>
    <p
      className="MuiFormHelperText-root-53"
      id="standard-full-width-helper-text"
    >
      After typing your string, please press Remove vowels button
    </p>
  </div>
  <button
    className="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-text-63 MuiButton-flat-66 Button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
    width="50px"
  >
    <span
      className="MuiButton-label-62"
    >
      Remove vowels
    </span>
    <span
      className="MuiTouchRipple-root-90"
    />
  </button>
  <p>
    Result: 
  </p>
  <p
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    
  </p>
</div>
`;
